packages/typespec-ts/test/modularUnit/scenarios/models/serialization/additionalProperties.md

```diff
@@ -28,60 +37,44 @@ compatibility-mode: true
 Generated Models.
 
 ```ts models
+import { serializeRecord } from "../static-helpers/serialization/serialize-record.js";
+
 /** model interface SimpleModel */
-export interface SimpleModel extends Record<string, string> {
+export interface SimpleModel extends Record<string, number | string> {
   propA: string;
   propB: string;
 }
 
 export function simpleModelSerializer(item: SimpleModel): any {
-  return { ...item, propA: item["propA"], propB: item["propB"] };
+  return {
+    ...serializeRecord(
+      item,
+      undefined,
+      _simpleModelAdditionalPropertySerializer,
```

Comment:
The value type transformation would also apply to legacy cases if there is ser/deser logic required.

Reply:
Revert this change caz we expose any to customers.